### PR TITLE
RGRIDT-900: Creating Filter by value and filter by condition actions

### PR DESCRIPTION
**Pull request by glundgren93**
This PR delivers filter by value and by condition functions.

### What was done
* Created API methods to filter by value and condition.
* Created methods on IColumnFilter and implemented them on ColumnFilter.

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/WijmoProvider/Features/ColumnFilter.ts [hunk 1]
```diff
@@ -108,6 +108,62 @@ namespace WijmoProvider.Feature {
             this.setState(this._enabled);
         }
 
+        public byCondition(
+            columnId: string,
+            values: OSFramework.OSStructure.FilterCondition[]
+        ): void {
+            const column = this._grid.getColumn(columnId);
+            if (column) {
+                const columnFilter = this._filter.getColumnFilter(
```
Comment (tmbp95) [inline code/src/WijmoProvider/Features/ColumnFilter.ts:117]: This line will work even if the column has the filter disabled, right?

**Review by tmbp95 — DISMISSED** [review]

File: code/src/WijmoProvider/Features/ColumnFilter.ts [hunk 2]
```diff
@@ -108,6 +108,62 @@ namespace WijmoProvider.Feature {
...
+                    column.config.binding
+                ).conditionFilter;
+
+                if (values.length > 0) {
+                    const condition1 = values[0];
+                    const condition2 = values[1];
+
+                    columnFilter.condition1.value =
+                        column.columnType === OSFramework.Enum.ColumnType.Number
+                            ? parseInt(condition1.value)
```
Comment (bmartinho) [inline code/src/WijmoProvider/Features/ColumnFilter.ts:127]: If column type is Number, the code is doing parseInt, shouldn't be for parseFloat? What about for Currency, Date and DateTime?

File: code/src/WijmoProvider/Features/ColumnFilter.ts (hunk 1 above)
Comment (glundgren93) [reply code/src/WijmoProvider/Features/ColumnFilter.ts:117]: Yes!

File: code/src/WijmoProvider/Features/ColumnFilter.ts (hunk 2 above)
Comment (glundgren93) [reply code/src/WijmoProvider/Features/ColumnFilter.ts]: You're right! Changed to parseFloat. Currency, date and datetime column expect a string.

**Review by rugoncalves — DISMISSED** [review]

**Review by tmbp95 — DISMISSED** [review]

**Pull request merged**